src/page/browser.js

```diff
@@ -0,0 +1,49 @@
+function isMinimumSupported() {
+  return (
+    'globalThis' in window &&
+    'Promise' in window &&
+    'allSettled' in Promise &&
+    'Symbol' in window &&
+    'replace' in Symbol &&
+    'WeakMap' in window &&
+    'assign' in Object &&
+    'entries' in Object &&
+    'values' in Object &&
+    'getOwnPropertyDescriptors' in Object &&
+    'fromEntries' in Object &&
+    'assign' in Object &&
+    'URL' in window &&
+    'toJSON' in URL.prototype &&
+    'URLSearchParams' in window &&
+    Array.prototype[Symbol.iterator] &&
+    'includes' in Array.prototype &&
+    'reduce' in Array.prototype &&
+    'sort' in Array.prototype &&
+    'flatMap' in Array.prototype &&
+    NodeList.prototype[Symbol.iterator] &&
+    'forEach' in NodeList.prototype &&
+    HTMLCollection.prototype[Symbol.iterator] &&
+    DOMTokenList.prototype[Symbol.iterator] &&
+    'forEach' in DOMTokenList.prototype &&
+    'fill' in Int8Array.prototype &&
+    'set' in Int8Array.prototype &&
+    'sort' in Int8Array.prototype &&
+    'replace' in String.prototype &&
+    'search' in String.prototype &&
+    'split' in String.prototype &&
+    'includes' in String.prototype &&
+    'match' in String.prototype &&
+    'trim' in String.prototype &&
+    'split' in String.prototype &&
+    'endsWith' in String.prototype &&
+    'replaceAll' in String.prototype &&
+    'sticky' in RegExp.prototype &&
+    'toString' in RegExp &&
+    parseFloat('1.23') === 1.23 &&
+    (1.23456789).toFixed(2) === '1.23'
+  );
+}
+
+if (!isMinimumSupported()) {
```

Comment:
wasn't it possible to move the async checks here as well? (cookies and indexeddb)?

Reply:
I would like to have more advanced code as a TS, then the types are clear, and I think some better syntax sugar (I'm not sure what pure JS offers, babel injects cool syntax features support on the fly). Here we just primitively check base functions availability and I like to keep it at a minimum. This will never be the code you'd like to debug. But if you see the value of having it here, I can try to move it.

Reply:
ah, I forgot: in the TS `CheckBrowser` I allow to run the browser for the oAuth case. I don't want to move oAuth to here, as it uses the constant `QUERY_KEY.SCOPE` - hardcoding it here would mean we must remember to fix it here too if some change happens in the main code. And I guess we don't need `IndexDB` or cookies for the oAuth  part, while things like `globalThis` and some other basics will be still required at least due to `sourceMap` injection by `webpack`